src/infrastructure/TransactionSearchCriteria.ts

```diff
@@ -0,0 +1,217 @@
+import { TransactionTypeEnum } from 'symbol-openapi-typescript-node-client/dist/model/transactionTypeEnum';
+import { Order } from 'symbol-openapi-typescript-node-client';
+
+/**
+ * Defines the params used to search transactions. With this criteria, you can sort and filter
+ * transactions queries using rest.
+ */
+export class TransactionSearchCriteria extends SearchCriteria {
+    /**
+     * Transaction identifier up to which transactions are returned. (optional)
+     */
+    private id?: string;
+
+    /**
+     * Filter by address involved in the transaction.
+     *
+     * An account's address is consider to be involved in the transaction when the account is the
+     * sender, recipient, or it is required to cosign the transaction.
+     *
+     * This filter cannot be combined with ''recipientAddress'' and ''signerPublicKey'' query
+     * params.  (optional)
+     */
+    private address?: Address;
+
+    /**
+     * Address of an account receiving the transaction. (optional)
+     */
+    private recipientAddress?: Address;
+
+    /**
+     * Public key of the account signing the entity. (optional)
+     */
+    private signerPublicKey?: string;
+
+    /**
+     * Filter by block height. (optional, default to null)
+     */
+    private height?: UInt64;
+
+    /**
+     * Entry id at which to start pagination. If the ordering parameter is set to -id, the elements
+     * returned precede the identifier. Otherwise, newer elements with respect to the id are
+     * returned.  (optional)
+     */
+    private offset?: string;
+
+    /**
+     * The group of transaction (optional, default is confirmed)
+     */
+    private group?: TransactionGroupSubsetEnum;
+
+    /**
+     * Filter by transaction type. To filter by multiple transaction type.  (optional, default to
+     * new empty array)
+     */
+    private transactionTypes?: TransactionType[];
+
+    /**
+     * When true, the endpoint also returns all the embedded aggregate transactions. When
+     * false, only top-level transactions used to calculate the block transactionsHash are
+     * returned.  (optional, default to false)
+     */
+    private embedded: boolean;
+
+    public getId(): string | undefined {
+        return this.id;
+    }
+
+    public getAddress(): string | undefined {
+        return this.address?.plain();
+    }
+
+    public getRecipientAddress(): string | undefined {
+        return this.recipientAddress?.plain();
+    }
+
+    public getSignerPublicKey(): string | undefined {
+        return this.signerPublicKey;
+    }
+
+    public getHeight(): string | undefined {
+        return this.height?.toString();
+    }
+
+    public getOffset(): string | undefined {
+        return this.offset;
+    }
+
+    public getGroup(): TransactionGroupSubsetEnum | undefined {
+        return this.group;
+    }
+
+    public getTransactionTypes(): TransactionTypeEnum[] | undefined {
+        return this.transactionTypes?.map((type) => type.valueOf());
```

Comment:
Move to the TransactionHttp, model should not know about TransactionTypeEum if possible